### PR TITLE
GT: Add device present PLDM event log

### PR DESCRIPTION
**Pull request by Quanta-YangChen**
Summary:
- Add device(E1.S, NIC) present event log.

Test Plan:
- Build Code: Pass

Dependency: #797

### CODE REVIEW

File: meta-facebook/gt-cc/src/platform/plat_isr.c [hunk 1]
```diff
@@ -106,3 +103,157 @@ void ISR_DC_ON()
...
+	LOG_WRN("HSC SMB is %s", is_alert ? "alert" : "non-alert");
+
+	if (pldm_send_platform_event(PLDM_SENSOR_EVENT, PLDM_EVENT_SENSOR_HSC,
+				     PLDM_STATE_SENSOR_STATE, (uint8_t *)&event,
+				     sizeof(struct pldm_sensor_event_state_sensor_state))) {
+		LOG_ERR("Send HSC SMB alert event log failed");
+	}
+}
+
+#define ISR_SSD_PRESENT_HANDLER(idx)                                                               \
```
Comment (garnermic) [inline meta-facebook/gt-cc/src/platform/plat_isr.c:222]: Is there a reason for all the line continuation characters in this function?

File: meta-facebook/gt-cc/src/platform/plat_pldm_monitor.c [hunk 2]
```diff
@@ -0,0 +1,320 @@
...
+	/* Avoid redundant workers sending */
+	if (k_work_delayable_busy_get(&data->work))
+		return;
+
+	k_work_schedule(&data->work, K_SECONDS(1));
+}
+
+void ssd_present_check()
+{
+	for (uint8_t i = 0; i < 16; i++) {
```
Comment (garnermic) [inline meta-facebook/gt-cc/src/platform/plat_pldm_monitor.c:286]: Please use a #define for this instead of a magic number.

**Review by garnermic — CHANGES_REQUESTED** [review]

File: meta-facebook/gt-cc/src/platform/plat_isr.c (hunk 1 above)
Comment (Quanta-YangChen) [reply meta-facebook/gt-cc/src/platform/plat_isr.c:222]: Hi @garnermic,

We use all the line continuation to define a multiple-line macro as a function those ones are the same but differ by the index for the SSD0 - SSD15. Is there anything risk we haven't considered? We can expand all those functions to normal function declaration and definition.

Thanks.

File: meta-facebook/gt-cc/src/platform/plat_pldm_monitor.c (hunk 2 above)
Comment (Quanta-YangChen) [reply meta-facebook/gt-cc/src/platform/plat_pldm_monitor.c]: Already update a [commit](https://github.com/facebook/OpenBIC/pull/798/commits/c23ed4e60b500713d5e8cc60f36168fa1ff736cf) modification as suggested. Please review it again, thank you.

**Comment by Quanta-YangChen** [comment]
Close this PR, and will resolve all checks in a later PR.

**Pull request closed without merging**